security/fma-isthmus-l2-pectra.md

```diff
@@ -0,0 +1,207 @@
+  - [Added input limits and specify acceleration for new precompiles](https://github.com/ethereum-optimism/specs/blob/a795bb1f8c34410c47f05d4c3614482361db64b1/specs/protocol/isthmus/exec-engine.md#evm-changes)
+- EIP-2935: Block hashes contract predeploy
+  - [Added network upgrade transaction](https://github.com/ethereum-optimism/specs/blob/a795bb1f8c34410c47f05d4c3614482361db64b1/specs/protocol/isthmus/derivation.md#eip-2935-contract-deployment)
+- EIP-7251: Increase the MAX_EFFECTIVE_BALANCE
+  - Only affects execution clients
+- EIP-7623: Increase calldata cost
+  - Only affects execution clients
+- EIP-7002: Execution layer triggerable withdrawals
+  - Not applicable to OP
+  - Requests sent to this contract are ignored, and the contract is not deployed by default
+- EIP-6110: Supply validator deposits on chain
+  - Not applicable to OP
+  - Requests sent to this contract are ignored, and the contract is not deployed by default
+- EIP-7685: General purpose execution layer requests
+  - No requests are currently applicable to OP, so the [execution requests array is always empty in Isthmus](https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/exec-engine.md#engine_newpayloadv4).
+
+## Failure Modes and Recovery Paths
+
+### FM1: Chain halt due to upgrade transaction failing
+
+- **Description:** The chain could halt if a network upgrade transaction can't be included in the upgrade block.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. end-to-end tests created for network upgrade transactions
+      - [Action test created here](https://github.com/ethereum-optimism/optimism/blob/01ddb2e6a09edf55a7cb2130e0a5b6acd0c2d2fa/op-e2e/actions/upgrades/isthmus_fork_test.go#L290)
+  2. same upgrade path as previous predeploys
+      - example: [EIP-4788 deployment transaction](https://github.com/ethereum-optimism/optimism/blob/b6b74290b5502f45daae57946f969327cdb2d383/op-node/rollup/derive/ecotone_upgrade_transactions.go#L130)
+    
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing the network upgrade transaction in rollup node client software (like `op-node`).
+
+### FM2: Chain halt due to request hash mismatch
+
+- **Description:** Requests hash was added to the block header. If it mismatches between the EL client and the rollup node, the chain could halt.
+- **Risk Assessment:** High severity, low likelihood
+- **Mitigations:**
+  1. End-to-end test ensuring no requests are included in blocks
+      - https://github.com/ethereum-optimism/optimism/pull/14253
+  2. Simplicity of validation rules - always must be an empty hash
+- **Detection:** Alert for L2 Unsafe liveness should be triggered
+- **Recovery Path(s)**: This would require an emergency update if it occurred fixing requests hash logic in respective clients (e.g. `op-node`, `op-geth`, `op-reth`, ...)
```

Comment:
How we make sure that the proxyd, op-conductor, op-batcher are also not impacted with the `request_hash` field.

Another question, seems to be mismatch hash so there are both real hash or they are empty hash both side? Because here, this is written that the hash should be empty (`always must be an empty hash`). 

1. If this is the case, how we can make sure that through P2P when we share block from another op-reth/op-geth this one will have a valid `request_hash`. 
2. Do we have test that ensure that if there is an incorrect `request_hash` propagated across multiples components.

Reply:
EDIT: On the specs I found this here we should link this to `request_hash`  + in the reference section. 
https://github.com/ethereum-optimism/specs/blob/b5e0fa98881171f658f782597a46b641e8f3dfd0/specs/protocol/isthmus/exec-engine.md#header-validity-rules

Reply:
Should be nice to update and add this PR for tests  in case of incorrect `Request_hash` header.

Reply:
Added those references. About the P2P issue, it reduces to peers following consensus rules. 

- If a malicious client sends a block header with requests root not set to `sha256('')`, the calculated requests root after EVM execution will not match the header and the block will be rejected.
- If a malicious client sends requests in the bodies, but still has an empty requests root, the requests in the bodies will not hash to the correct requests root.

I can add these to a separate FMA if you want, or include them here.

Reply:
@meyer9 yes, I do think we should put here please and then we can resolve this!